src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java

```diff
@@ -74,24 +74,34 @@
     private final Secret secretKey;
 
     private final String iamRoleArn;
+    private final String iamExternalId;
     private final String iamMfaSerialNumber;
 
     private volatile Integer stsTokenDuration;
 
     // Old data bound constructor. It is maintained to keep binary compatibility with clients that were using it directly.
     public AWSCredentialsImpl(@CheckForNull CredentialsScope scope, @CheckForNull String id,
                               @CheckForNull String accessKey, @CheckForNull String secretKey, @CheckForNull String description) {
-        this(scope, id, accessKey, secretKey, description, null, null);
+        this(scope, id, accessKey, secretKey, description, null, null, null);
     }
 
-    @DataBoundConstructor
+    // Old data bound constructor. It is maintained to keep binary compatibility with clients that were using it directly.
     public AWSCredentialsImpl(@CheckForNull CredentialsScope scope, @CheckForNull String id,
                               @CheckForNull String accessKey, @CheckForNull String secretKey, @CheckForNull String description,
                               @CheckForNull String iamRoleArn, @CheckForNull String iamMfaSerialNumber) {
+        this(scope, id, accessKey, secretKey, description, iamRoleArn, iamMfaSerialNumber, null);
+    }
+
+    @DataBoundConstructor
+    public AWSCredentialsImpl(@CheckForNull CredentialsScope scope, @CheckForNull String id,
+                              @CheckForNull String accessKey, @CheckForNull String secretKey, @CheckForNull String description,
+                              @CheckForNull String iamRoleArn, @CheckForNull String iamMfaSerialNumber,
+                              String iamExternalId) {
         super(scope, id, description);
         this.accessKey = Util.fixNull(accessKey);
         this.secretKey = Secret.fromString(secretKey);
         this.iamRoleArn = Util.fixNull(iamRoleArn);
+        this.iamExternalId = Util.fixNull(iamExternalId);
```

Comment:
is there a reason for converting a null field to `""`. externalId is optional from AWS perspective. sending an empty string can trigger validation issues at AWS:
```com.amazonaws.services.securitytoken.model.AWSSecurityTokenServiceException: 1 validation error detected: Value '' at 'externalId' failed to satisfy constraint: Member must have length greater than or equal to 2 (Service: AWSSecurityTokenService; Status Code: 400; Error Code: ValidationError; Request ID:```